airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -107,17 +114,40 @@ def get(self, url: str, params: Mapping = None):
         response = self._session.get(self._api_prefix + url, params=params)
         return self._parse_and_handle_errors(response)
 
+    def consume_credit(self, credit):
```

Comment:
This logic is complex. Couldn't we make this a lot simpler by just keeping two numbers: `current_time_window_start` and `current_window_credits_consumed` then running something like: 

```
if not self._requests_per_minute: 
  return 



# Reset time window if it has elapsed
if time.time() > self.current_time_window_start + 60:
  self.current_time_window_start = time.time()
  self.current_window_credits_consumed = 0

if self.current_window_credits_consumed >= self._requests_per_minute
  logger.trace(f"Reached call limit for this minute, wait for {sleep_time:.2f} seconds")
  sleep(max(1, 60 - (time.time() - self.current_window_start_time))
```

or is there an advantage to the current approach

Reply:
ok, I like your approach, original logic was taken from a python client,
and I was sure that it is fine to handle calls with the same price, but it is not good enough for calls where the price could be 3, 5, etc